src/archive.rs

```diff
@@ -425,7 +430,7 @@ impl<'a> EntriesFields<'a> {
                  size in header",
             ));
         }
-        entry.size = cur;
+        entry.header.set_size(cur);
```

Comment:
I sort of forget where this is all called from, but this `set_size` feels not quite right here because it's rewriting the header that users may be inspecting. I think it would be best if this `cur` is still saved off in a `tar`-crate-specific location.

Reply:
This is called at the end of reading `next_entry()`. It was was originally stored at the `EntryFields` `size` field, but since that is no longer present, where is the best location given we do not want to create any additional fields?

Reply:
It may be the case that my initial inclination to not duplicate the information in a few places may not have been accurate. It may be that we need to bring back a `size` field in the struct

Reply:
Hmm, yeah I think so. I'll make the changes this week.

Reply:
I've reverted the changes back to using the same `size` field. I do think it is cleanest currently while the sparse files need to modify the `size`.